src/ConstructiveSolidGeometry/Intervals.jl

```diff
@@ -15,4 +15,80 @@
+
+@inline _in_angular_interval_closed(α::Real, α_int::Nothing) = true
+@inline _in_angular_interval_closed(α::Real, α_int::Nothing, tol::Real) = true
+@inline _in_angular_interval_closed(α::Real, α_int::AbstractInterval{T}) where {T} = mod(α - α_int.left, T(2π)) ≤ α_int.right - α_int.left
+@inline _in_angular_interval_closed(α::Real, α_int::AbstractInterval{T}, tol::Real) where {T} = mod(α - (α_int.left-tol), T(2π)) ≤ (α_int.right+tol) - (α_int.left-tol)
+
+@inline _in_angular_interval_open(α::T, α_int::Nothing) where {T<:Real} = 0 < mod(α, T(2π)) < 2π
+@inline _in_angular_interval_open(α::Real, α_int::AbstractInterval{T}) where {T} = 0 < mod(α - α_int.left, T(2π)) < α_int.right - α_int.left
+
+_in_angular_interval_union(val::T, α::AbstractInterval, β::AbstractInterval) where {T<:AbstractFloat} =
+    _in_angular_interval_closed(val, α) || _in_angular_interval_closed(val, β)
+
+function _is_edge_of_angular_interval_union(val::T, α::AbstractInterval, β::AbstractInterval) where {T<:AbstractFloat}
+    tol = 10*geom_atol_zero(T)
+    (_in_angular_interval_union(val+tol, α, β) && !_in_angular_interval_union(val-tol, α, β)) ||
+    (_in_angular_interval_union(val-tol, α, β) && !_in_angular_interval_union(val+tol, α, β))
+ end
+
+function union_angular_intervals(α::AbstractInterval{T}, β::AbstractInterval{T}) where {T} #if no intersection will return nothing
+    tol = 10*geom_atol_zero(T)
+    if α == β
+        return α
+    elseif α.left == α.right && _in_angular_interval_closed(α.left, β)
+        return β
+    elseif β.left == β.right && _in_angular_interval_closed(β.left, α)
+        return α
+    elseif α.left ≥ 0 && β.left ≥ 0 && α.right < T(2π) && β.right < T(2π) #less not less or equal to 2pi
+        return isempty(α ∩ β) ? nothing : α ∪ β
+    else
+        edges = 0
+        θ1 = T(0)
+        θ2 = T(0)
+        if _is_edge_of_angular_interval_union(α.left, α, β)
+            θ1 = α.left
+            edges = edges + 1
+        end
+        if _is_edge_of_angular_interval_union(α.right, α, β)
+            edges == 0 ? θ1 = α.right : θ2 = α.right
+            edges = edges + 1
+        end
+        if _is_edge_of_angular_interval_union(β.left, α, β) && mod(β.left, T(2π)) ≠ mod(α.left, T(2π)) && mod(β.left, T(2π)) ≠ mod(α.right, T(2π))
+            if edges == 0
+                θ1 = β.left
+            elseif edges == 1
+                θ2 = β.left
+            end
+            edges = edges + 1
+        end
+        if edges < 3 && _is_edge_of_angular_interval_union(β.right, α, β) && mod(β.right, T(2π)) ≠ mod(α.left, T(2π)) && mod(β.right, T(2π)) ≠ mod(α.right, T(2π))
+            edges == 1 ? θ2 = β.right : nothing
+            edges = edges + 1
+        end
+        if edges == 2
+            θMin, θMax = minmax(mod(θ1, T(2π)), mod(θ2, T(2π)))
+            return _in_angular_interval_union(θMin + tol, α, β) ? θMin..θMax : (θMax-T(2π))..θMin
+        elseif edges == 0
+            return T(0)..T(2π)
+        end
```

Comment:
I should have put some comments. If there is a contiguous union it returns the resulting interval (even if it is `T(0)..T(2π)`). When there is no contiguous union it returns nothing (edges different than 2 or 0). In cases like this for non periodic intervals IntervalSets.jl returns an error.

Reply:
This is breaking a bit what we have for the volume primitives:
For the volume primitives, an "interval" of `nothing` means full `2π`.
Here, `nothing` describes an empty interval. 
Can we avoid this inconsistency somehow?

Also, I only see this method is only called once in your code, checking whether the output is `nothing` (in `is_intersection_an_interval`), where you could also just call `isempty(α ∩ β)`. Or am I missing something?

Is this function important for the future because I do not see the need for it now?